Enable more linters: `ineffassign`, `unconvert`, `misspell`

awly opened the pull request:
All changes are purely cosmetic, except for `lib/srv/regular/sshserver.go` - this appears to be an actual bug fix.
Looks like we never advertised the custom listening port (`aport`), if it differed from regular listening port (`port`).

Updates #3551

Commit:
Fix common misspellings detected by `misspell` linter

Commit:
Remove unnecessary type conversions

Caught by `unconvert` linter. No behavior changes here.

Commit:
Remove ineffectual assignments

Caught by `ineffassign` linter.

Note the change in lib/srv/regular/sshserver.go - looks like we
advertised the custom listening port (`aport`), if it differed from
regular listening port (`port`).

Commit:
Update e-ref

Pick up `misspell` fixes.

Commit:
Enable new linters: `ineffassign`, `unconvert`, `misspell`

All findings have been fixed and `make lint` returns no errors.

russjones commented on lib/srv/regular/sshserver.go:
```diff
@@ -623,7 +623,7 @@ func (s *Server) AdvertiseAddr() string {
 	if aport == "" {
 		aport = port
 	}
-	return net.JoinHostPort(ahost, port)
```
Can you add a test, that will also help clear up expected behavior here.

awly replied:
Done, also renamed `advertiseIP` to `advertiseAddr` to better reflect what it is.

Commit:
Expand tests for sshserver.AdvertiseAddr

Also, rename advertiseIP to advertiseAddr to better represent what it
contains. It can have a custom port, not just IP.

Refactoring the rest of the codebase to use "advertiseAddr" instead of
"advertiseIP" (and *utils.NetAddr instead of string) is too noisy and
low value for this commit.
